internal/events/eventstream.go

```diff
@@ -231,9 +254,25 @@ func (a *eventStream) preUpdateStream() error {
 	close(a.updateInterrupt)
 	a.batchCond.Broadcast()
 	a.batchCond.L.Unlock()
+
+	a.drainBlockConfirmationManager()
+
 	return nil
 }
 
+func (a *eventStream) drainBlockConfirmationManager() {
```

Comment:
should we purge the cache here?

Reply:
It's intentional we don't here, because the drain is only scoped to one event stream.
Whereas the cache is across all event streams, based on blocks.